packages/overlay/src/integrations/sentry/hooks/useDebounce.ts

```diff
@@ -1,17 +1,27 @@
-import { useEffect, useState } from 'react';
+// Ref: https://stackoverflow.com/a/78529642/16866418
+import { useLayoutEffect, useMemo, useRef } from 'react';
 
-export default function useDebounce<T>(value: T, delay: number): T {
-  const [debouncedValue, setDebouncedValue] = useState<T>(value);
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export default function useDebounce<T extends (...args: any[]) => void>(callback: T, delay: number) {
+  const callbackRef = useRef(callback);
 
-  useEffect(() => {
-    const handler = setTimeout(() => {
-      setDebouncedValue(value);
-    }, delay);
+  useLayoutEffect(() => {
+    callbackRef.current = callback;
+  }, [callback]);
 
-    return () => {
-      clearTimeout(handler);
-    };
-  }, [value, delay]);
+  let timer: number;
 
-  return debouncedValue;
+  const debounceFunction = (func: T, delayMs: number, ...args: Parameters<T>) => {
+    clearTimeout(timer);
+    timer = window.setTimeout(() => {
+      func(...args);
+    }, delayMs);
+  };
```

Comment:
May also use this instead: https://www.npmjs.com/package/lodash.debounce

Reply:
Yes this would be the simpler and the best solution here 😅  
what do you think instead of hook, should we use this package?

Reply:
See my other comment, will try to use this and see how it goes